--- a/include/hw/xen/xen.h	2015-01-21 23:54:36.856666662 -0500
+++ b/include/hw/xen/xen.h	2015-01-21 23:55:39.356666667 -0500
@@ -28,6 +28,15 @@
     return xen_allowed;
 }
 
+static inline int xen_stubdom_enable(void)
+{
+#ifdef CONFIG_STUBDOM
+    return 1;
+#else
+    return 0;
+#endif
+}
+
 int xen_pci_slot_get_pirq(PCIDevice *pci_dev, int irq_num);
 void xen_piix3_set_irq(void *opaque, int irq_num, int level);
 void xen_piix_pci_write_config_client(uint32_t address, uint32_t val, int len);
