--- a/qemu-seccomp.c
+++ b/qemu-seccomp.c
@@ -50,6 +50,7 @@ static const struct QemuSeccompSyscall b
     { SCMP_SYS(syslog),                 QEMU_SECCOMP_SET_DEFAULT },
     { SCMP_SYS(mount),                  QEMU_SECCOMP_SET_DEFAULT },
     { SCMP_SYS(umount),                 QEMU_SECCOMP_SET_DEFAULT },
+    { SCMP_SYS(umount2),                QEMU_SECCOMP_SET_DEFAULT },
     { SCMP_SYS(kexec_load),             QEMU_SECCOMP_SET_DEFAULT },
     { SCMP_SYS(afs_syscall),            QEMU_SECCOMP_SET_DEFAULT },
     { SCMP_SYS(break),                  QEMU_SECCOMP_SET_DEFAULT },
@@ -60,6 +61,11 @@ static const struct QemuSeccompSyscall b
     { SCMP_SYS(mpx),                    QEMU_SECCOMP_SET_DEFAULT },
     { SCMP_SYS(prof),                   QEMU_SECCOMP_SET_DEFAULT },
     { SCMP_SYS(profil),                 QEMU_SECCOMP_SET_DEFAULT },
+    { SCMP_SYS(kcmp),                   QEMU_SECCOMP_SET_DEFAULT },
+    { SCMP_SYS(perf_event_open),        QEMU_SECCOMP_SET_DEFAULT },
+    { SCMP_SYS(process_vm_readv),       QEMU_SECCOMP_SET_DEFAULT },
+    { SCMP_SYS(process_vm_writev),      QEMU_SECCOMP_SET_DEFAULT },
+    { SCMP_SYS(ptrace),                 QEMU_SECCOMP_SET_DEFAULT },
     { SCMP_SYS(putpmsg),                QEMU_SECCOMP_SET_DEFAULT },
     { SCMP_SYS(security),               QEMU_SECCOMP_SET_DEFAULT },
     { SCMP_SYS(stty),                   QEMU_SECCOMP_SET_DEFAULT },
@@ -72,6 +78,7 @@ static const struct QemuSeccompSyscall b
     { SCMP_SYS(bdflush),                QEMU_SECCOMP_SET_OBSOLETE },
     { SCMP_SYS(create_module),          QEMU_SECCOMP_SET_OBSOLETE },
     { SCMP_SYS(get_kernel_syms),        QEMU_SECCOMP_SET_OBSOLETE },
+    { SCMP_SYS(nfsservctl),             QEMU_SECCOMP_SET_OBSOLETE },
     { SCMP_SYS(query_module),           QEMU_SECCOMP_SET_OBSOLETE },
     { SCMP_SYS(sgetmask),               QEMU_SECCOMP_SET_OBSOLETE },
     { SCMP_SYS(ssetmask),               QEMU_SECCOMP_SET_OBSOLETE },
@@ -93,6 +100,7 @@ static const struct QemuSeccompSyscall b
     { SCMP_SYS(fork),                   QEMU_SECCOMP_SET_SPAWN },
     { SCMP_SYS(vfork),                  QEMU_SECCOMP_SET_SPAWN },
     { SCMP_SYS(execve),                 QEMU_SECCOMP_SET_SPAWN },
+    { SCMP_SYS(execveat),               QEMU_SECCOMP_SET_SPAWN },
     /* resource control */
     { SCMP_SYS(getpriority),            QEMU_SECCOMP_SET_RESOURCECTL },
     { SCMP_SYS(setpriority),            QEMU_SECCOMP_SET_RESOURCECTL },
