--- a/hw/xen/xen_pt_config_init.c
+++ b/hw/xen/xen_pt_config_init.c
@@ -1434,6 +1434,7 @@ static XenPTRegInfo xen_pt_emu_reg_msi[]
 };
 
 
+#ifndef CONFIG_STUBDOM
 /**************************************
  * MSI-X Capability
  */
@@ -1528,6 +1529,7 @@ static XenPTRegInfo xen_pt_emu_reg_msix[
         .size = 0,
     },
 };
+#endif
 
 static XenPTRegInfo xen_pt_emu_reg_igd_opregion[] = {
     /* Intel IGFX OpRegion reg */
@@ -1663,6 +1665,7 @@ static int xen_pt_msi_size_init(XenPCIPa
     *size = msi_size;
     return 0;
 }
+#ifndef CONFIG_STUBDOM
 /* get MSI-X Capability Structure register group size */
 static int xen_pt_msix_size_init(XenPCIPassthroughState *s,
                                  const XenPTRegGroupInfo *grp_reg,
@@ -1680,6 +1683,7 @@ static int xen_pt_msix_size_init(XenPCIP
     *size = grp_reg->grp_size;
     return 0;
 }
+#endif
 
 
 static const XenPTRegGroupInfo xen_pt_emu_reg_grps[] = {
@@ -1773,6 +1777,7 @@ static const XenPTRegGroupInfo xen_pt_em
         .size_init   = xen_pt_pcie_size_init,
         .emu_regs = xen_pt_emu_reg_pcie,
     },
+#ifndef CONFIG_STUBDOM
     /* MSI-X Capability Structure reg group */
     {
         .grp_id      = PCI_CAP_ID_MSIX,
@@ -1781,6 +1786,7 @@ static const XenPTRegGroupInfo xen_pt_em
         .size_init   = xen_pt_msix_size_init,
         .emu_regs = xen_pt_emu_reg_msix,
     },
+#endif
     /* Intel IGD Opregion group */
     {
         .grp_id      = XEN_PCI_INTEL_OPREGION,
