--- a/hw/i386/xen/xen-hvm.c
+++ b/hw/i386/xen/xen-hvm.c
@@ -1273,6 +1273,10 @@ void xen_hvm_init(PCMachineState *pcms,
     state->wakeup.notify = xen_wakeup_notifier;
     qemu_register_wakeup_notifier(&state->wakeup);
 
+    if (xen_stubdom_enable()) {
+        xc_set_hvm_param(xen_xc, xen_domid, HVM_PARAM_DM_DOMAIN, DOMID_SELF);
+    }
+
     rc = xen_get_ioreq_server_info(xen_domid, state->ioservid,
                                    &ioreq_pfn, &bufioreq_pfn,
                                    &bufioreq_evtchn);
@@ -1372,12 +1376,21 @@ void xen_hvm_init(PCMachineState *pcms,
     state->device_listener = xen_device_listener;
     device_listener_register(&state->device_listener);
 
+#ifndef CONFIG_STUBDOM
     /* Initialize backend core & drivers */
     if (xen_be_init() != 0) {
         error_report("xen backend core setup failed");
         goto err;
     }
     xen_be_register_common();
+#else
+    xenstore = xs_daemon_open();
+    if (!xenstore) {
+        error_report("can't connect to xenstored");
+        goto err;
+    }
+#endif
+
     xen_read_physmap(state);
 
     /* Disable ACPI build because Xen handles it */
