--- a/configure
+++ b/configure
@@ -310,6 +310,7 @@
 libusb=""
 usb_redir=""
 glx=""
+stubdom="no"
 zlib="yes"
 lzo=""
 snappy=""
@@ -1050,6 +1051,8 @@
   ;;
   --enable-usb-redir) usb_redir="yes"
   ;;
+  --enable-stubdom) stubdom="yes"
+  ;;
   --disable-zlib-test) zlib="no"
   ;;
   --disable-lzo) lzo="no"
@@ -1374,6 +1377,7 @@
   --enable-usb-redir       enable usb network redirection support
   --enable-lzo             enable the support of lzo compression library
   --enable-snappy          enable the support of snappy compression library
+  --enable-stubdom         enable building the ioemu-stubdom
   --disable-guest-agent    disable building of the QEMU Guest Agent
   --enable-guest-agent     enable building of the QEMU Guest Agent
   --with-vss-sdk=SDK-path  enable Windows VSS support in QEMU Guest Agent
@@ -4339,6 +4343,7 @@
 if test "$xen" = "yes" ; then
   echo "xen ctrl version  $xen_ctrl_version"
 fi
+echo "stubdom support   $stubdom"
 echo "brlapi support    $brlapi"
 echo "bluez  support    $bluez"
 echo "Documentation     $docs"
@@ -4782,6 +4787,11 @@
   echo "RBD_LIBS=$rbd_libs" >> $config_host_mak
 fi
 
+# stubdom support
+if test "$stubdom" = "yes"; then
+  echo "CONFIG_STUBDOM=y" >> $config_host_mak
+fi
+
 echo "CONFIG_COROUTINE_BACKEND=$coroutine" >> $config_host_mak
 if test "$coroutine_pool" = "yes" ; then
   echo "CONFIG_COROUTINE_POOL=1" >> $config_host_mak
