--- a/configure
+++ b/configure
@@ -357,6 +357,7 @@ rbd=""
 smartcard=""
 libusb=""
 usb_redir=""
+stubdom="no"
 opengl=""
 opengl_dmabuf="no"
 cpuid_h="no"
@@ -1164,6 +1165,8 @@ for opt do
   ;;
   --enable-usb-redir) usb_redir="yes"
   ;;
+  --enable-stubdom) stubdom="yes"
+  ;;
   --disable-zlib-test) zlib="no"
   ;;
   --disable-lzo) lzo="no"
@@ -1516,6 +1519,7 @@ disabled with --disable-FEATURE, default
   usb-redir       usb network redirection support
   lzo             support of lzo compression library
   snappy          support of snappy compression library
+  enable-stubdom  support for Xen stubdom
   bzip2           support of bzip2 compression library
                   (for reading bzip2-compressed dmg images)
   seccomp         seccomp support
@@ -5340,6 +5344,7 @@ if test "$xen" = "yes" ; then
   echo "xen ctrl version  $xen_ctrl_version"
   echo "pv dom build      $xen_pv_domain_build"
 fi
+echo "stubdom support   $stubdom"
 echo "brlapi support    $brlapi"
 echo "bluez  support    $bluez"
 echo "Documentation     $docs"
@@ -5889,6 +5894,11 @@ if test "$rbd" = "yes" ; then
   echo "RBD_LIBS=$rbd_libs" >> $config_host_mak
 fi
 
+# stubdom support
+if test "$stubdom" = "yes"; then
+  echo "CONFIG_STUBDOM=y" >> $config_host_mak
+fi
+
 echo "CONFIG_COROUTINE_BACKEND=$coroutine" >> $config_host_mak
 if test "$coroutine_pool" = "yes" ; then
   echo "CONFIG_COROUTINE_POOL=1" >> $config_host_mak
