From 61cd574f29f41046f1c709cfa9da118156babf83 Mon Sep 17 00:00:00 2001
From: Anthony PERARD <anthony.perard@citrix.com>
Date: Fri, 1 Jun 2012 15:47:01 +0100
Subject: [PATCH 2/2] fix/remap_area_mfn_pte_fn

---
 arch/x86/xen/mmu.c | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/arch/x86/xen/mmu.c b/arch/x86/xen/mmu.c
index 69f5857..999fc82 100644
--- a/arch/x86/xen/mmu.c
+++ b/arch/x86/xen/mmu.c
@@ -2606,7 +2606,20 @@ static int remap_area_mfn_pte_fn(pte_t *ptep, pgtable_t token,
 				 unsigned long addr, void *data)
 {
 	struct remap_data *rmd = data;
-	pte_t pte = pte_mkspecial(mfn_pte(rmd->mfn++, rmd->prot));
+
+	/* Use the native_make_pte function because we are sure we don't
+	 * have to do any pfn->mfn translations but at the same time we
+	 * could in a stubdom so xen_initial_domain() would return false. */
+	pte_t pte = pte_mkspecial(native_make_pte(((phys_addr_t)(rmd->mfn++) << PAGE_SHIFT)
+	                                          | massage_pgprot(rmd->prot)));
+	pteval_t val = pte_val_ma(pte);
+
+#if 0
+	if (pat_enabled && !WARN_ON(val & _PAGE_PAT)) {
+		if ((val & (_PAGE_PCD | _PAGE_PWT)) == _PAGE_PWT)
+			val = (val & ~(_PAGE_PCD | _PAGE_PWT)) | _PAGE_PAT;
+	}
+#endif
 
 	rmd->mmu_update->ptr = virt_to_machine(ptep).maddr;
 	rmd->mmu_update->val = pte_val_ma(pte);
-- 
Anthony PERARD

